protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/fs/UriResourceFetcher.java

```diff
@@ -148,26 +148,29 @@ private Optional<Path> handleOtherUri(URI uri, String extension) throws Resoluti
       conn.connect();
 
       try (
-          var responseStream = new BufferedInputStream(conn.getInputStream());
-          var fileStream = new SizeAwareBufferedOutputStream(Files.newOutputStream(targetFile))
+          var responseInputStream = new BufferedInputStream(conn.getInputStream());
+          var fileOutputStream = FileUtils.newBufferedOutputStream(targetFile)
       ) {
-        responseStream.transferTo(fileStream);
-        log.info("Downloaded '{}' to '{}' ({} bytes)", uri, targetFile, fileStream.size);
+        responseInputStream.transferTo(fileOutputStream);
       }
 
-      return Optional.of(targetFile);
-
-    } catch (FileNotFoundException ex) {
-
-      // May be raised during the call to .getInputStream(), or the call to .connect(),
-      // depending on the implementation.
-      log.warn("No resource at '{}' appears to exist!", uri);
+      var fileSize = Files.size(targetFile);
+      log.info("Downloaded '{}' to '{}' ({} bytes)", uri, fileSize);
 
-      return Optional.empty();
+      return Optional.of(targetFile);
 
     } catch (IOException ex) {
       log.debug("Failed to download '{}' to '{}'", uri, targetFile, ex);
-      throw new ResolutionException("Failed to download '" + uri + "' to '" + targetFile + "'", ex);
+
+      if (ex instanceof FileNotFoundException) {
```

Comment:
moved into IOException catch block so we can log the FNFE to the debug log stream rather than just discarding the stacktrace. Useful for diagnostics if users raise any bugs relating to this in the future.